bot/exts/events/advent_of_code/_cog.py

```diff
@@ -185,24 +186,61 @@ async def join_leaderboard(self, ctx: commands.Context) -> None:
     @whitelist_override(channels=AOC_WHITELIST_RESTRICTED)
-    async def aoc_leaderboard(self, ctx: commands.Context) -> None:
-        """Get the current top scorers of the Python Discord Leaderboard."""
+    async def aoc_leaderboard(
+            self,
+            ctx: commands.Context,
+            day_and_star: Optional[str] = None,
+            maximum_scorers: Optional[int] = 10
+    ) -> None:
+        """
+        Get the current top scorers of the Python Discord Leaderboard.
+
+        Additionally, you can provide a day-star(e.g. 2-1, as second day first star) to filter your query more.
+        """
+        if maximum_scorers > AocConfig.max_day_and_star_results or maximum_scorers <= 0:
+            raise commands.BadArgument(
+                f"The maximum number of results you can query is {AocConfig.max_day_and_star_results}"
+            )
         async with ctx.typing():
             try:
                 leaderboard = await _helpers.fetch_leaderboard()
-            except _helpers.FetchingLeaderboardFailed:
+            except _helpers.FetchingLeaderboardFailedError:
                 await ctx.send(":x: Unable to fetch leaderboard!")
                 return
+            if not day_and_star:
 
-            number_of_participants = leaderboard["number_of_participants"]
+                number_of_participants = leaderboard["number_of_participants"]
 
-            top_count = min(AocConfig.leaderboard_displayed_members, number_of_participants)
-            header = f"Here's our current top {top_count}! {Emojis.christmas_tree * 3}"
+                top_count = min(AocConfig.leaderboard_displayed_members, number_of_participants)
+                header = f"Here's our current top {top_count}! {Emojis.christmas_tree * 3}"
 
-            table = f"```\n{leaderboard['top_leaderboard']}\n```"
-            info_embed = _helpers.get_summary_embed(leaderboard)
+                table = f"```\n{leaderboard['top_leaderboard']}\n```"
+                info_embed = _helpers.get_summary_embed(leaderboard)
+
+                await ctx.send(content=f"{header}\n\n{table}", embed=info_embed)
+                return
 
-            await ctx.send(content=f"{header}\n\n{table}", embed=info_embed)
+            # This is a dictionary that contains solvers in respect of day, and star.
+            # e.g. 1-1 will fetch the solvers of the first star of the first day and their completion time
+            per_day_and_star = json.loads(leaderboard['leaderboard_per_day_and_star'])
+            try:
+                scorers = per_day_and_star[day_and_star][:maximum_scorers]
+            except KeyError:
+                raise commands.BadArgument(
+                    "Day and star format has a notation of 'day-star' day being 1 to 25, and star is either 1 or "
+                    "2 (e.g. 3-1) "
+                )
+
+            day, star = day_and_star.split("-")
+            codeblock = f"```csharp\nTop scorers for day {day}, star {star} in UTC timezone\n"
```

Comment:
What exactly does the `csharp` bit do here? What importance does the coloring for csharp have?

Reply:
It highlights the data in the codeblock, and I find it a perfect fit for this case. Take a look at the image sent below.

Reply:
I see. all right lgtm then!